src/main/java/net/snowflake/ingest/streaming/internal/OpenChannelRequestInternal.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import net.snowflake.ingest.utils.Constants;
+
+/** Class used to serialize the {@link OpenChannelRequest} */
+class OpenChannelRequestInternal implements StreamingIngestRequest {
+  @JsonProperty("request_id")
+  private String requestId;
+
+  @JsonProperty("role")
+  private String role;
+
+  @JsonProperty("channel")
+  private String channel;
+
+  @JsonProperty("table")
+  private String table;
+
+  @JsonProperty("database")
+  private String database;
+
+  @JsonProperty("schema")
+  private String schema;
+
+  @JsonProperty("write_mode")
+  private String writeMode;
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
```

Comment:
I'd remove this NON_NULL and add a conditional to the callsite (left another comment there).
That way you have a consistent json serialization behavior across all contracts instead of this one-off

Reply:
Discussed offline. Keep `@JsonInclude(JsonInclude.Include.NON_NULL)`